Data/Binary/Serialise/CBOR/Term.hs

```diff
@@ -102,7 +102,8 @@ decodeTerm :: Decoder Term
 decodeTerm = do
     tkty <- peekTokenType
     case tkty of
-      TypeUInt   -> fromWord <$> decodeWord
+      TypeUInt   -> do w <- decodeWord
+                       return $! fromWord w
```

Comment:
If it does't already exist, why not make a combinator for this:

```haskell
(<$!>) :: Monad f => (a -> b) -> f a -> f b
g <$!> fa = do a <- fa; return $! g a
```
Then the changes would be minor.

Also, would there be any advantage in making it `!a <- fa...`?